pkg/workflows/sdk/v2/promise.go

```diff
@@ -0,0 +1,58 @@
+package sdk
+
+import (
+	"sync"
+)
+
+type Promise[T any] interface {
+	Await() (T, error)
+	CapabilityPromise
+}
+
+type CapabilityPromise interface {
```

Comment:
what pattern is this/what's the value add where a private no-op method is embedded?